src/main/scheduler/scheduler.h

```diff
@@ -18,68 +18,99 @@
 #pragma once
 
 //#define SCHEDULER_DEBUG
+//#define SCHEDULER_DEBUG_PRINT
 
 typedef enum {
-    TASK_PRIORITY_IDLE = 0,     // Disables dynamic scheduling, task is executed only if no other task is active this cycle
-    TASK_PRIORITY_LOW = 1,
+    TASK_PRIORITY_IDLE = 0,         // This task will only run if there is nothing else to run. There is no guarantee this will ever run if the cpu is always under load!
+    TASK_PRIORITY_LOW = 1,          
     TASK_PRIORITY_MEDIUM = 3,
     TASK_PRIORITY_HIGH = 5,
-    TASK_PRIORITY_REALTIME = 6,
+    TASK_PRIORITY_REALTIME = 6,     // This task will be ran as close to realtime as possible. This will aggressively preempt any other task.
     TASK_PRIORITY_MAX = 255
 } cfTaskPriority_e;
 
+// A special task type that will reference the current task.
 #define TASK_SELF -1
 
 typedef struct {
     const char * taskName;
-    bool         isEnabled;
-    uint32_t     desiredPeriod;
-    uint8_t      staticPriority;
-    uint32_t     maxExecutionTime;
-    uint32_t     totalExecutionTime;
-    uint32_t     averageExecutionTime;
-    uint32_t     latestDeltaTime;
+    bool         isEnabled;                 // Indicates if the task is enabled.
+    uint32_t     desiredPeriod;             // the desired amount of time between each execution (in microseconds)
+    uint8_t      priority;                  // the priority of the task
+    bool         isWaitingToBeRan;          // Indicates if the task is waiting to be ran or not.
```

Comment:
'ran' is past tense, so 'isWaitingToBeRun'  but perhaps 'pending' might be simpler?